flink-core/src/main/java/org/apache/flink/configuration/NettyShuffleEnvironmentOptions.java

```diff
@@ -173,6 +173,28 @@
 				" help relieve back-pressure caused by unbalanced data distribution among the subpartitions. This value should be" +
 				" increased in case of higher round trip times between nodes and/or larger number of machines in the cluster.");
 
+	/**
+	 * Maximum number of network buffers can be used per sort-merge blocking result partition.
+	 */
+	@Documentation.Section(Documentation.Sections.ALL_TASK_MANAGER_NETWORK)
+	public static final ConfigOption<Integer> NETWORK_MAX_BUFFERS_PER_SORT_MERGE_PARTITION =
+		key("taskmanager.network.sort-merge-blocking-shuffle.max-buffers-per-partition")
+			.defaultValue(2048)
+			.withDescription("Maximum number of network buffers can be used per sort-merge blocking result partition. " +
+				"This value is only an upper bound limit and does not mean that the sort-merge blocking result partition" +
+				" will use as many network buffers.");
+
+	/**
+	 * Parallelism threshold to switch between sort-merge based blocking shuffle and the default hash-based blocking shuffle.
+	 */
+	@Documentation.Section(Documentation.Sections.ALL_TASK_MANAGER_NETWORK)
+	public static final ConfigOption<Integer> NETWORK_SORT_MERGE_SHUFFLE_MIN_PARALLELISM =
+		key("taskmanager.network.sort-merge-blocking-shuffle.min-parallelism")
+			.defaultValue(Integer.MAX_VALUE)
```

Comment:
Please add the `intType()` here to make it runtime type safe (and not have the deprecation warnings).